subprojects/signing/src/main/java/org/gradle/plugins/signing/Sign.java

```diff
@@ -96,6 +97,12 @@ private Stream<Signature> signatureStream() {
     public Sign() {
         // If we aren't required and don't have a signatory then we just don't run
         onlyIf("Signing is required, or signatory is set", spec(task -> isRequired() || getSignatory() != null));
+
+        getProject().getTasks().withType(AbstractPublishToMaven.class).forEach(
+            abstractPublishToMaven -> {
+                abstractPublishToMaven.dependsOn(this);
```

Comment:
I believe this is not a proper fix, but a workaround. This introduces task dependencies that should not be normally required.
As a consequence, for example, any publishing task would be able to run only after all signing tasks are completed, that means publishing artifacts would be shifted to the end of the project build only when all artifacts are built and signed.

Reply:
What would be your suggestion to fix this? 🤔

I guess [this code snipped](https://github.com/gradle/gradle/issues/26091#issuecomment-1798137734) is a bit better in that regards, right?

As it uses configureEach it will only set the direct task dependencies of the tasks that are really required to run, if it understand it correctly.

Reply:
The problem as I understand it is that there are multiple `Sign` tasks signing the same artifact file and producing the same signature file, so that when the corresponding publishing task uses the output of one such Sign task, it gets an implicit dependency on all other Sign tasks.

Therefore, to solve it properly:
- either there should be no multiple signing tasks that sign the same file and produce the same signature file;
- or each signing task should produce a distinctly named signature file, even when applied to the same input artifact file.